omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            representing each obstacle
+
+    Returns:
+        th.Tensor: (N, 4)-shaped tensor, where each of the N computed collision spheres are defined by (x,y,z,r),
+            where (x,y,z) is the global position and r defines the sphere radius
+    """
+    n_obstacles = len(obstacles)
+    if not isinstance(n_spheres, Iterable):
+        n_spheres = [n_spheres] * n_obstacles
+    if not isinstance(sphere_radius, Iterable):
+        sphere_radius = [sphere_radius] * n_obstacles
+
+    sph_list = []
+    for obs, n_sph, sph_radius in zip(obstacles, n_spheres, sphere_radius):
+        sph = obs.get_bounding_spheres(
+            n_sph,
+            sph_radius,
+            pre_transform_pose=lazy.curobo.types.math.Pose(
+                position=th.zeros(3), quaternion=th.tensor([1.0, 0, 0, 0])
+            ).to(tensor_args),
+            tensor_args=tensor_args,
+            fit_type=lazy.curobo.geom.sphere_fit.SphereFitType.VOXEL_VOLUME_SAMPLE_SURFACE,
+            voxelize_method="ray",
+        )
+        sph_list += [s.position + [s.radius] for s in sph]
+
+    return tensor_args.to_device(th.as_tensor(sph_list))
+
+
+def get_sphere_representation(
+    obj, tensor_args, reference_prim_path=None, n_spheres=100, sphere_radius=0.001, extra_links=None, n_extra_spheres=20
+):
+    """
+    Gest the collision sphere representation of object @obj as well as any additional links specified by @extra_links
+
+    Args:
+        obj (BaseObject or None): Non-articulated object whose sphere representation will be computed. If None, will
+            not use any object
+        tensor_args (TensorDeviceType): Tensor device information
+        reference_prim_path (None or str): If specified, the reference prim path from which all sphere poses will
+            be computed
+        n_spheres (int): Number of collision spheres for representing @obj
+        sphere_radius (float): Radius of the generated collision spheres
+        extra_links (None or list of RigidPrim): If specified, any additional links which should be included
+            in the sphere representation
+        n_extra_spheres (int): Number of collision spheres for representing each additional link specified in
+            @extra_links
+
+    Returns:
+        th.Tensor: (N, 4)-shaped tensor, where each of the N computed collision spheres are defined by (x,y,z,r),
+            where (x,y,z) is the position and r defines the sphere radius taken wrt obj.prim_path
+    """
+    obstacles = []
+    n_spheres_total = []
+
+    if obj is not None:
+        target_obstacle = get_obstacles(
+            reference_prim_path=reference_prim_path,
+            only_substring=[obj.prim_path],
+            ignore_substring=["collision", "container"],
+        )
+        assert (
+            len(target_obstacle.objects) == 1
+        ), f"Found multiple target obstacles, but expected 1. Got: {[obj.name for obj in target_obstacle.objects]}"
```

Comment:
we should add some comment to explain `target_obstacle.objects` are a list of visual meshes

Reply:
removed